### PR TITLE
Add proximis installer

### PR DESCRIPTION
**Pull request by maxhelias**
I need feedback before updating the README

### CODE REVIEW

File: project.sh [hunk 1]
```diff
@@ -0,0 +1,176 @@
+#!/bin/bash
+
+# Load constante
```
Comment (matthieupetel) [inline project.sh]: "constant" instead of "constante"

File: project.sh [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+            echo "${info}No install process found.${reset}"
+        fi
+
+        if [ "$?" = "0" ]; then
+            echo "${success}Project $INTEGRATOR / $REMOTE was successfully installed.${reset}"
+        else
+            echo "${error}Something went wrong. Project installation may have failed.${reset}"
+        fi
+    elif echo "$answer" | grep -iq "^n" ;then
+        echo "${info}Do not forget to runt the Proximis update.sh script or the make command later.${reset}"
```
Comment (matthieupetel) [inline project.sh]: "run" instead of "runt"

File: project.sh [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+    rm -rf $PROJECT_DIR
+    git clone ssh://git@gitlab.omn.proximis.com:10022/$INTEGRATOR/$PROJECT.git $PROJECT_DIR
+
+    # Add forked remote
+    git --git-dir=$GIT_DIR --work-tree=$GIT_WORK_TREE checkout -b dev origin/dev
+    git --git-dir=$GIT_DIR --work-tree=$GIT_WORK_TREE remote add composer ssh://git@gitlab.omn.proximis.com:10022/$REMOTE/$PROJECT.git
+    git --git-dir=$GIT_DIR --work-tree=$GIT_WORK_TREE fetch composer
+
+    # TODO : Add project.local.json in a better way
+    git clone git@bitbucket.org:snippets/emakinafr-commerce/re8zR5/projectlocaljson.git $PROJECT_DIR/App/Config/project-local-json
```
Comment (ajardin) [inline project.sh]: As the snippet is private, you could add the template in the repository.

File: project.sh [hunk 4]
```diff
@@ -0,0 +1,178 @@
+#!/bin/bash
```
Comment (ajardin) [inline project.sh]: Could you try to put the snippet below instead?

```
#!/usr/bin/env bash
set -euo pipefail
```

File: project.sh [hunk 5]
```diff
@@ -0,0 +1,178 @@
...
+
+function update() {
+    echo "${info}Update started...${reset}"
+
+    # Remove all plugins
+    rm -rf $PROJECT_DIR/Plugins/*
+
+    # Update composer
+    composer self-update
+    composer update --working-dir=$PROJECT_DIR --prefer-dist --ignore-platform-reqs
```
Comment (ajardin) [inline project.sh]: Is it safe enough to perform an update with the `--ignore-platform-reqs` parameter? Maybe you can run the command inside the container instead.

File: Makefile [hunk 6]
```diff
@@ -39,6 +39,29 @@ uninstall: ## Uninstall the environment
 
 .PHONY: build env start stop restart install uninstall
 
+##
+## ----------------------------------------------------------------------------
+##   Proximis
```
Comment (ajardin) [inline Makefile:44]: Your targets may be safer if you put `&&` operators between commands instead of `;`.

File: project.sh (hunk 5 above)
Comment (maxhelias) [reply project.sh]: We do it to be able to feed the plugin folder and after install plugin from GitLab

File: project.local.json [hunk 7]
```diff
@@ -0,0 +1,106 @@
...
+        "redis": {
+          "server": {
+            "host": "redis",
+            "port": "6379"
+          }
+        }
+      }
+    },
+    "Db": {
+      "cryptKey": "XXXX",
```
Comment (ker0x) [inline project.local.json]: You can remove this key as it will be automatically added when installing the project

File: project.local.json [hunk 8]
```diff
@@ -0,0 +1,106 @@
...
+      "auth": true,
+      "connection": "login",
+      "username": "XXXX",
+      "password": "XXXX",
+      "ssl": "ssl",
+      "fakemail": [
+      ]
+    },
+    "Svi": {
+      "fakeSvi": "XXXX"
```
Comment (ker0x) [inline project.local.json]: I'm not in favor of leaving keys with undefined values

File: project.local.json [hunk 9]
```diff
@@ -0,0 +1,106 @@
...
+    },
+    "Sms": {
+      "fakeSms": "XXXX"
```
Comment (ker0x) [inline project.local.json]: Same here

**Review by ker0x — APPROVED** [review]

**Review by matthieupetel — APPROVED** [review]

File: Makefile (hunk 6 above)
Comment (ajardin) [reply Makefile:44]: I think you can also put `&& \` between the second and the third instruction. :-)

**Review by ajardin — APPROVED** [review]

**Pull request merged**